test/compiler/codegen.jl

```diff
@@ -702,6 +702,23 @@ function f42645()
 end
 @test ((f42645()::B42645).y::A42645{Int}).x
 
+struct A44921{T}
+    x::T
+end
+function f44921(a)
+    if a == :x
+        A44921(_f) # _f purposefully undefined
+    elseif a == :p
+        g44921(a)
+    end
+end
+function g44921(a)
+    if !any([])
```

Comment:
this can be any unknown condition, so might be simplest to make it `_f` also?

Reply:
Since this code path is actually run, not sure how we can use `_f` here without getting an `UndefVarError`?

Reply:
ah, true, would need to be something like `!@isdefined _f`